common/k8s/src/lease.rs

```diff
@@ -0,0 +1,123 @@
+use chrono::Utc;
+use k8s_openapi::api::coordination::v1::Lease;
+use k8s_openapi::apimachinery::pkg::apis::meta::v1::MicroTime;
+use kube::api::{Api, ListParams, Patch, PatchParams};
+use kube::core::ObjectList;
+use kube::Client;
+use serde::{Deserialize, Serialize};
+
+pub const K8S_STARTUP_LEASE_LABEL: &str = "process=agent-startup";
+pub const K8S_STARTUP_LEASE_RETRY_ATTEMPTS: i32 = 3;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct LeasePatchSpec {
+    spec: LeasePatchValue,
+}
+
+// Non snake case for JSON parsing reasons
+#[allow(non_snake_case)]
+#[derive(Debug, Serialize, Deserialize)]
+struct LeasePatchValue {
+    holderIdentity: Option<String>,
```

Comment:
We can use serde's rename mechanism so that we can use snake case here

Reply:
Updated.